### PR TITLE
main: command line option for socks5 proxy

### PR DESCRIPTION
**Pull request by selsta**

### CODE REVIEW

File: pages/settings/SettingsLayout.qml [hunk 1]
```diff
@@ -265,16 +266,17 @@ Rectangle {
 
         MoneroComponents.RemoteNodeEdit {
             id: proxyEdit
+            enabled: socksProxyFlag.length === 0
             Layout.leftMargin: 36
             Layout.topMargin: 6
             Layout.minimumWidth: 100
             placeholderFontSize: 15
-            visible: persistentSettings.proxyEnabled
+            visible: persistentSettings.proxyEnabled || socksProxyFlag.length > 0
```
Comment (xiphon) [inline pages/settings/SettingsLayout.qml:274]: Use `proxyCheckbox.checked`

Comment (selsta) [reply pages/settings/SettingsLayout.qml]: done, also updated the enabled part

File: pages/settings/SettingsLayout.qml [hunk 2]
```diff
@@ -265,16 +266,17 @@ Rectangle {
 
         MoneroComponents.RemoteNodeEdit {
             id: proxyEdit
+            enabled: proxyCheckbox.enabled
```
Comment (xiphon) [inline pages/settings/SettingsLayout.qml:269]: Shouldn't allow changing the proxy address if socksProxyFlag is set.

File: main.qml [hunk 3]
```diff
@@ -1423,13 +1423,14 @@ ApplicationWindow {
         property string proxyAddress: "127.0.0.1:9050"
         property bool proxyEnabled: false
         function getProxyAddress() {
-            if (!proxyEnabled) {
+            var proxyAddressSetOrForced = socksProxyFlagSet ? socksProxyFlag : proxyAddress;
```
Comment (xiphon) [inline main.qml:1426]: Could be declared at line 1430

File: main.qml [hunk 4]
```diff
@@ -1423,13 +1423,14 @@ ApplicationWindow {
...
+            if (socksProxyFlagSet ? !socksProxyFlag : !proxyEnabled) {
```
Comment (xiphon) [inline main.qml:1427]: IMO it would be better to not use ternary operator here

**Review by xiphon — APPROVED** [review]

**Pull request merged**